test/helpers.bash

```diff
@@ -502,6 +502,7 @@ function temp_sandbox_conf() {
 
 function wait_until_exit() {
 	ctr_id=$1
+	expected_exit_status=$2
```

Comment:
You could simplify your diff here with a fallback during [parameter expansion][1]:

```bash
expected_exit_status="${2:-0}"
```

which would allow you to drop your current `"0"` additions and only set `"1"` when you needed it.

An alternative would be to pass this in as an environment variable instead of a positional argument, which would allow:

```bash
EXPECTED_EXIT_STATUS=1 wait_until_exit "$ctr_id"
```

and then in the function:

```bash
[[ "$output" =~ "Exit Code: ${EXPECTED_EXIT_STATUS:-0}" ]]
```

Personally, I prefer the more-explicit environment-variable approach, but I'm not a maintainer.

[1]: http://pubs.opengroup.org/onlinepubs/9699919799/utilities/V3_chap02.html#tag_18_06_02

Reply:
I liked the idea of the environment variable and made the modification, thanks for the advice.